agent/src/beerocks/slave/agent_db.cpp

```diff
@@ -0,0 +1,96 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2019-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "agent_db.h"
+
+#include <easylogging++.h>
+
+using namespace beerocks;
+
+AgentDB::sRadio *AgentDB::radio(const std::string &iface_name)
+{
+    if (iface_name.empty()) {
+        LOG(ERROR) << "Given radio iface name is empty";
+        return nullptr;
+    }
+
+    auto radio_it = std::find_if(m_radios.begin(), m_radios.end(), [&](const sRadio &radio_entry) {
+        return iface_name == radio_entry.front.iface_name ||
+               iface_name == radio_entry.back.iface_name;
+    });
+
+    return radio_it == m_radios.end() ? nullptr : &(*radio_it);
+}
+
+bool AgentDB::add_radio(const std::string &front_iface_name, const std::string &back_iface_name)
+{
+    if (front_iface_name.empty() && back_iface_name.empty()) {
+        LOG(ERROR) << "Both front and back ifaces names are empty!";
+        return false;
+    }
+
+    if (!front_iface_name.empty()) {
+        if (radio(front_iface_name)) {
+            LOG(DEBUG) << "Radio entry of front iface " << front_iface_name
+                       << " is already exist. Ignore.";
+            return true;
+        }
+    }
```

Comment:
@mariomaz has a point, this even improves readability:
```suggestion
    if (!front_iface_name.empty() && radio(front_iface_name)) {
        LOG(DEBUG) << "Radio entry of front iface " << front_iface_name
                   << " already exists. Ignore.";
        return true;
    }
```

Reply:
fixed